syntax = "proto3";

package flight;

service UserService {
  rpc CheckUser (CheckUserRequest) returns (CheckUserResponse) {}
  rpc GetUser (GetUserRequest) returns (GetUserResponse) {}
}

message CheckUserRequest {
  string email = 1;
}

message CheckUserResponse {
  bool exists = 1;
}

message GetUserRequest {
  string email = 1;
}

message GetUserResponse {
  string email = 1;
  string name = 2;
  string surname = 3;
  string fiscal_code = 4;
  string bank_info = 5;
  bool found = 6;
}

service CollectorService {
  rpc CleanupUser (CleanupUserRequest) returns (CleanupUserResponse) {}
  rpc RemoveAirportInterest (RemoveAirportInterestRequest) returns (RemoveAirportInterestResponse) {}
}

message CleanupUserRequest {
  string email = 1;
}

message CleanupUserResponse {
  bool success = 1;
  string message = 2;
}

message RemoveAirportInterestRequest {
  string email = 1;
  string airport_code = 2;
}

message RemoveAirportInterestResponse {
  bool success = 1;
  string message = 2;
}

